File: backend/src/main/java/de/garrafao/phitag/domain/judgement/spanjudgement/SpanJudgement.java
package de.garrafao.phitag.domain.judgement.spanjudgement;

import de.garrafao.phitag.domain.annotator.Annotator;
import de.garrafao.phitag.domain.instance.spaninstance.SpanInstance;
import de.garrafao.phitag.domain.judgement.IJudgement;
import lombok.Getter;
import lombok.Setter;
import org.apache.commons.lang3.Validate;

import javax.persistence.*;

@Entity
@Table(name = "phitagspanjudgement")
@Getter
public class SpanJudgement implements IJudgement {

    @EmbeddedId
    private SpanJudgementId id;

    @MapsId("annotatorid")
    @ManyToOne
    private Annotator annotator;

    @MapsId("instanceid")
    @ManyToOne
    private SpanInstance instance;

    @Setter
    @Column(name = "label", nullable = false)
    private String label;

    @Setter
    @Column(name = "comment", nullable = false)
    private String comment;

    public SpanJudgement() {
    }

    public SpanJudgement(final SpanInstance instance, final Annotator annotator, final String label, final String comment) {
        Validate.notNull(instance, "instance must not be null");
        Validate.notNull(annotator, "annotator must not be null");
        // Validate.notBlank(label, "label must not be null");

        this.id = new SpanJudgementId(instance.getId(), annotator.getId());

        this.instance = instance;
        this.annotator = annotator;

        this.label = label;
        this.comment = comment;
    }
    
}

File: frontend/lib/model/judgement/spanjudgement/model/SpanJudgement.ts
import LexSubInstance from "../../../instance/lexsubinstance/model/LexSubInstance";
import SpanInstance from "../../../instance/spaninstance/model/SpanInstance";
import { IJudgement, IJudgementConstructor } from "../../model/IJudgement";
import SpanJudgementDto from "../dto/SpanJudgementDto";
import SpanJudgementId from "./SpanJudgementId";

export default class SpanJudgement implements IJudgement {

    readonly id: SpanJudgementId;

    readonly instance: SpanInstance;

    readonly label: string;
    readonly comment: string;

    constructor(id: SpanJudgementId, instance: SpanInstance, label: string, comment: string) {
        this.id = id;
        this.instance = instance;
        this.label = label;
        this.comment = comment;
    }

    public getId(): SpanJudgementId {
        return this.id;
    }

    public getInstance(): SpanInstance {
        return this.instance;
    }

    public getLabel(): string {
        return this.label;
    }

    public getComment(): string {
        return this.comment;
    }

    public static fromDto(dto: SpanJudgementDto): SpanJudgement {
        return new SpanJudgement(
            SpanJudgementId.fromDto(dto.id),
            SpanInstance.fromDto(dto.instance),
            dto.label,
            dto.comment
        );
    }

}

export class SpanJudgementConstructor implements IJudgementConstructor {
    fromDto(dto: SpanJudgementDto): SpanJudgement {
        return SpanJudgement.fromDto(dto);
    }
}